File: LeetCode/Day 3/Solution.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int strStr(string haystack, string needle) {
        int L = haystack.length(), n = needle.length(), m = L;
        for (int i = 0; i <= m - n; i++) {
            if (haystack.substr(i, n) == needle) {
                return i;
            }
        }
        return -1;
    }
};

int main() {
    string haystack = "sadbutsad";
    string needle = "sad";

    Solution solution;
    int index = solution.strStr(haystack, needle);

    cout << "Index of needle: " << index << endl;

    return 0;
}

File: LeetCode/Day 3/Solution.py
class Solution():
    def strStr(self, haystack: str, needle: str) -> int:
            n, m = len(needle), len(haystack)
            for i in range(m - n + 1):
                if haystack[i:i + n] == needle:
                    return i
            return -1
    

# Example 1
haystack = "sadbutsad"
needle = "sad"
output = Solution().strStr(haystack, needle)
print(output)  # Output: 0

# Example 2
haystack = "leetcode"
needle = "leeto"
output = Solution().strStr(haystack, needle)
print(output)  # Output: -1
